server/pulp/server/db/model/dispatch.py

```diff
@@ -614,7 +616,9 @@ def save(self, fields_to_set_on_insert=None):
         update = {'$set': stuff_to_update}
         if set_on_insert:
             update['$setOnInsert'] = set_on_insert
-        self.get_collection().update(
-            {'task_id': task_id},
-            update,
-            upsert=True)
+        if mongoengine.get_version()  < "0.9.0":
+            self.get_collection().update({'task_id': task_id}, update, upsert=True,
+                                         write_option={'w':1, 'fsync':True})
+        else:
+            self.get_collection().update({'task_id': task_id}, update, upsert=True,
+                                         write_concern={'w':1, 'fsync':True})
```

Comment:
If we drop safe, the above becomes:

```
if mongoengine.get_version()  < "0.9.0":
    self.get_collection().update({'task_id': task_id}, update, upsert=True, write_option={'w':1, 'fsync':True})
else:
    self.get_collection().update({'task_id': task_id}, update, upsert=True, write_concern={'w':1, 'fsync':True})
```

There is repetition in there to achieve write_option vs. write_concern. What if you did this instead:

```
wc_param = 'write_option' if mongoengine.get_version()  < "0.9.0" else 'write_concern'
update_kwargs = {wc_param: {'w':1, 'fsync':True}}

self.get_collection().update({'task_id': task_id}, update, upsert=True, **update_kwargs)
```